packages/ui/components/form/select/SelectDownshift.tsx

```diff
@@ -0,0 +1,248 @@
+        case useCombobox.stateChangeTypes.ItemClick:
+        case useCombobox.stateChangeTypes.InputBlur:
+          return {
+            ...changes,
+            ...(changes.selectedItem && { isOpen: true, highlightedIndex: changes.highlightedIndex }),
+          };
+        default:
+          return changes;
+      }
+    },
+    onStateChange({ type, selectedItem: newSelectedItem, inputValue: newInputValue }) {
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+          if (newSelectedItem) {
+            if (selectedItems.includes(newSelectedItem)) {
+              removeSelectedItem(newSelectedItem);
+              props.onSelectedItemsChange && props.onSelectedItemsChange(selectedItems);
+              return;
+            }
+            const newSelectedItems = [...selectedItems, newSelectedItem];
+            setSelectedItems(newSelectedItems);
+            props.onSelectedItemsChange && props.onSelectedItemsChange(newSelectedItems);
+          }
+          break;
+
+        case useCombobox.stateChangeTypes.InputChange:
+          setInputValue(newInputValue || "");
+
+        default:
+          break;
+      }
+    },
+  });
+
+  return (
+    <>
+      <div className="flex flex-col">
+        <Label {...getLabelProps()}>Pick some books:</Label>
+        <div
+          aria-label="menu-toggle"
+          {...getToggleButtonProps(getDropdownProps({ preventKeyAction: isOpen }))}
+          className="flex h-9 space-x-1 rounded-md border border-gray-300 p-1 focus-within:border-gray-900 hover:cursor-pointer hover:border-gray-400 focus-visible:border-gray-900">
+          {!props.renderNode
+            ? selectedItems.map(function renderSelectedItem(selectedItemForRender, index) {
+                return (
+                  <div
+                    className="hocus:text-gray-900 hocus:bg-gray-300 focus:border-1 flex items-center rounded-md border-transparent bg-gray-200 px-2 py-[6px] text-sm text-gray-700"
+                    key={`selected-item-${index}`}
+                    {...getSelectedItemProps({
+                      selectedItem: selectedItemForRender,
+                      index,
+                    })}>
+                    <span>
+                      <>{selectedItemForRender[props.labelKey]}</>
+                    </span>
+                    <span
+                      className="ml-2 cursor-pointer"
+                      onClick={(e) => {
+                        e.stopPropagation();
+                        removeSelectedItem(selectedItemForRender);
+                      }}>
+                      &#10005;
+                    </span>
+                  </div>
+                );
+              })
+            : props.renderNode(selectedItems, getSelectedItemProps, removeSelectedItem)}
+        </div>
+      </div>
+      {/* @TODO: Make this UL (Button part) overrideable via props  */}
+      <ul
```

Comment:
It would be cool if we somehow could inject custom children in this select. 
By default you inject `<MultiSelect.List>`, which uses a context/zustand to get the data perhaps? And if you want to customise it, you can render your own component that uses the same hook to get the context, and render your own thing? 🤔 Makes the multiselect at least a lot easier from a props perspective. Just give it a thought 😁

Reply:
And looking a bit more into detail, it's perhaps a few more components instead of only one.